### PR TITLE
Global Contribution Types test

### PR DESCRIPTION
**Pull request by jranks123**
## Why are you doing this?
As a follow up to https://github.com/guardian/support-frontend/pull/1207/files, we want to run a global contributions types test. Although we ran this already in the US, we will rerun it to give us further confidence in the results. 

We will analyse the test from 2 perspectives, US and non-US (combining GB, AU and ROW)

The variants are: 

|Control|
|-------|
|image|

|Default Annual|
|-------|
|image|

|Default Annual no monthly|
|-------|
|image|

### CODE REVIEW

File: assets/helpers/checkouts.js [hunk 1]
```diff
@@ -72,7 +73,15 @@ function getValidContributionTypesFromUrlOrElse(fallback: ContributionType[]): C
   return fallback;
 }
 
-function getValidContributionTypes(): ContributionType[] {
+function getValidContributionTypes(abParticipations: Participations): ContributionType[] {
+  const { globalContributionTypes } = abParticipations;
+  const params = globalContributionTypes
+    ? globalContributionTypes.split('_')
```
Comment (tomrf1) [inline assets/helpers/checkouts.js]: can this bit just be `if (globalContributionTypes === 'default-annual_no-monthly') return ['ONE_OFF', 'ANNUAL']`?

Comment (jranks123) [reply assets/helpers/checkouts.js]: 👍

File: assets/helpers/checkouts.js [hunk 2]
```diff
@@ -72,7 +73,11 @@ function getValidContributionTypesFromUrlOrElse(fallback: ContributionType[]): C
   return fallback;
 }
 
-function getValidContributionTypes(): ContributionType[] {
+function getValidContributionTypes(abParticipations: Participations): ContributionType[] {
+  const { globalContributionTypes } = abParticipations;
+  if (globalContributionTypes === 'default-annual_no-monthly') {
+    return ['ONE_OFF', 'ANNUAL'];
+  }
   return getValidContributionTypesFromUrlOrElse(['ONE_OFF', 'MONTHLY', 'ANNUAL']);
```
Comment (joelochlann) [inline assets/helpers/checkouts.js]: personally I'd make the test one the `OrElse` - not sure if anyone's using the URL mechanism yet but I'd expect it to override tests

File: assets/pages/new-contributions-landing/contributionsLandingInit.js [hunk 3]
```diff
@@ -51,15 +53,24 @@ function getInitialPaymentMethod(
...
-function getInitialContributionType(): ContributionType {
-  const contributionType: ContributionType =
-    getContributionTypeFromUrlOrElse(getContributionTypeFromSessionOrElse('MONTHLY'));
+function getInitialContributionType(abParticipations: Participations): ContributionType {
+  const { globalContributionTypes } = abParticipations;
+  const abTestParams = globalContributionTypes
+    ? globalContributionTypes.split('_')
+    : [];
+
+  let contributionType: ContributionType;
```
Comment (joelochlann) [inline assets/pages/new-contributions-landing/contributionsLandingInit.js]: again, I'd probably do

```javascript
const fallback = abTestParams.includes('default-annual') ? 'ANNUAL' : 'MONTHLY';
contributionType = getContributionTypeFromUrlOrElse(getContributionTypeFromSessionOrElse(fallback));
```

File: assets/helpers/abTests/abtestDefinitions.js [hunk 4]
```diff
@@ -34,4 +34,21 @@ export const tests: Tests = {
...
+    ],
+    audiences: {
+      ALL: {
+        offset: 0,
+        size: 1,
+      },
+    },
+    isActive: true,
+    independent: true,
+    seed: 5,
```
Comment (joelochlann) [inline assets/helpers/abTests/abtestDefinitions.js]: to your point about excluding people from the test who have URL overrides: let's check this in the `canRun`

**Pull request merged**